### ISSUE TITLE
Include name in latest log file nodename

### ISSUE THREAD

**Issue by bjarnef** [DanDiplo/UmbracoTraceLogViewer#30]
In v `2.4.0` where it list latest version of log files as nodes, can the nodename be more unique. E.g. when having a log file for both `Courier` and `Umbraco` for same day, both nodenames are the same.

![image](https://user-images.githubusercontent.com/2919859/31493726-ada16da0-af50-11e7-95a4-2b7f3b41a6ed.png)

![image](https://user-images.githubusercontent.com/2919859/31493742-c340fc0c-af50-11e7-9e8a-b035d41e0409.png)

Can you include filename or for Umbraco and Courier just `UmbracoTraceLog` and `CoruierTraceLog` ... or strip the names to `Umbraco` and `Courier`?

**Comment by DanDiplo** [comment]
Hi Bryan. You can click on the Filenames folder to browse by filename, so you can distinguish. Does that help?

**Comment by bjarnef** [comment]
Yes, I know. I just think it will be better to distinguish the nodenames for each log. If you have other packages or custom stuff, you can end up with several nodes with same name.

**Comment by DanDiplo** [comment]
The bit you are looking at just shows the logs for that day and utilises date. I could show the filenames there, but filenames tend to be very long and hard to read. Now the viewer supports all sorts of different log filenames then trying to parse out things like machine name etc. becomes impossible - there's a lot of different formats they can be. Your example is just one case, but it has to work with all sorts of other names that people might use. That's why I added the alternative option of browsing by filename for the (small) number of cases where people might have more than one log for each day (as below).

![filetable](https://user-images.githubusercontent.com/6676168/31498025-cc9aa552-af58-11e7-8dde-fafd3cdda816.PNG)

**Comment by bjarnef** [comment]
Yes, but I think you can handle the values for Umbraco and Courier - if they starts with og just contains "UmbracoTraceLog" or "CourierTraceLog". I have seen these file names have changed since they were included in Umbraco.

**Comment by DanDiplo** [comment]
The problem is that this is just catering for the smallish number of people who use Courier. For everyone else they'd see 2017-01-05 (Umbraco) against _every_ node, which wouldn't be useful (and I'd get complaints, no doubt!). And it wouldn't work for people who have customised the naming in log4net. So at the moment it seems like a corner case.

So it's something I'll bear in mind, but currently I'm not planning on making this chance so it can be easily achieved by browsing by filename.

**Comment by bjarnef** [comment]
Well, with Umbraco Cloud more people will be using Courier (now Deploy, but log file is still named Courier).

I only meant for the top level nodes, where there only is one per type of log file.
![image](https://user-images.githubusercontent.com/2919859/31506820-de44b322-af78-11e7-84fb-23ebcb2d7c64.png)

Maybe there are alternatives to distinguish between the nodes at root level like adding an number/index as prefix or create an overview node similar to listview.

**Comment by DanDiplo** [comment]
I think there needs to be something more general implemented to distinguish files, as it doesn't just apply to Courier - so I need a solution that works for everyone. (For instance, Azure will often swap machines so you get multiple logs per day, which is another similar issue). But any "fix" needs to work for all these scenarios. I'll think about what can be done. Thanks.

**Comment by DanDiplo** [comment]
OK, I've released a new version that somewhat addresses these requests: https://our.umbraco.org/projects/developer-tools/diplo-trace-log-viewer/

It will now display the machine name next to the date eg. `2017-11-11 (BF-T440P-W10)` if the machine name of the log file differs from the machine of the host viewing the file.

If a file is a Courier log (ie. filename begins with "Courier") then it appends a `(C) `suffix after the date.

You can also sort and search by machine name in the table view.

**Comment by bjarnef** [comment]
@DanDiplo I am using the latest version 2.4.1 and UmbracoTraceLog I also have some other custom logging CustomLog, ProductImportLog and ProductIndexLog.

This is how it looks in backoffice.

![image](https://user-images.githubusercontent.com/2919859/41810753-da54dbf4-7703-11e8-8aa8-8b7489613c25.png)

Although I can click each of these four nodes and see which one of the logging files it is, it would be great if the tree names in some way were different.

**Comment by DanDiplo** [comment]
This was why I added the Filenames folder so that you could browse by specific filenames. I don't really want the date name to get much longer, as it gets messy, and not many people have multiple files, so it's a bit of an edge case.
